web/tests/test_self_report_attendance.py

```
@@ -0,0 +1,198 @@
+        """Test that students cannot report attendance for future sessions"""
+        self.client.login(username="student", password="studentpass")
+
+        response = self.client.post(reverse("self_report_attendance", args=[self.future_session.id]))
+
+        # Should redirect to course detail page
+        self.assertEqual(response.status_code, 302)
+
+        # Should not create attendance record
+        self.assertFalse(SessionAttendance.objects.filter(session=self.future_session, student=self.student).exists())
+
+    def test_already_reported_pending(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_already_reported_pending`

Add return type annotation: `None`

(ANN201)